pkg/runtime/wfengine/backends/actors/activity_actor.go

```diff
@@ -138,7 +140,10 @@ func (a *activityActor) InvokeReminder(ctx context.Context, reminder actors.Inte
 	timeoutCtx, cancelTimeout := context.WithTimeout(ctx, a.defaultTimeout)
 	defer cancelTimeout()
 
-	err := a.executeActivity(timeoutCtx, reminder.Name, state.EventPayload)
+	completed, err := a.executeActivity(timeoutCtx, reminder.Name, state.EventPayload)
+	if completed == runCompletedTrue {
+		a.completed.Store(true)
```

Comment:
completed will be forever true once there has been one reminder execution that set this to true.... dont you need some logic to make it go to false again?

or what are the semantics of this new `Completed() bool` function that you added?

Reply:
Correct- because workflows (and activities) reach a terminal state, they will be set as `Completed` 'forever'. When checked by the actor runtime, this being true if what makes this struct be deleted from the internal actors map and garbage collected. Without this, we have the situation today where internal actor state (workflow history etc.) is never garbage collected and the memory leak occurs.